diff --git a/src/client/ns_turn_msg.c b/src/client/ns_turn_msg.c
index d0c8d88..3319536 100644
--- a/src/client/ns_turn_msg.c
+++ b/src/client/ns_turn_msg.c
@@ -104,7 +104,7 @@ long turn_random(void)
 static void turn_random_tid_size(void *id)
 {
 	uint32_t *ar=(uint32_t*)id;
-	if(!RAND_pseudo_bytes((unsigned char *)ar,12)) {
+	if(!RAND_bytes((unsigned char *)ar,12)) {
 		size_t i;
 		for(i=0;i<3;++i) {
 			ar[i] = (uint32_t)random();
