From: =?utf-8?b?TcOpc3rDoXJvcyBNaWjDoWx5?= <misi@majd.eu>
Date: Sat, 2 Mar 2019 23:26:55 +0100
Subject: Set logging to syslog
Forwarded: not-needed

---
 examples/etc/turnserver.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/examples/etc/turnserver.conf
+++ b/examples/etc/turnserver.conf
@@ -526,7 +526,7 @@
 
 # Option to redirect all log output into system log (syslog).
 #
-#syslog
+syslog
 
 # Set syslog facility for syslog messages
 # Default values is ''.
